src/main/java/org/andersen/homework/model/entity/ticket/Ticket.java

```diff
@@ -1,33 +1,56 @@
 package org.andersen.homework.model.entity.ticket;
 
+import jakarta.persistence.Column;
+import jakarta.persistence.DiscriminatorColumn;
+import jakarta.persistence.DiscriminatorType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Inheritance;
+import jakarta.persistence.InheritanceType;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
 import java.math.BigDecimal;
 import java.util.UUID;
 import lombok.AllArgsConstructor;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
-import lombok.ToString;
-import org.andersen.homework.annotation.NullableWarning;
+import org.andersen.homework.model.entity.user.Client;
 
+@Entity
+@Table(name = "ticket")
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
```

Comment:
Why do we need here Inheritance annotation? Why exact this InheritanceType?

Reply:
I have Ticket as parent class, ConcertTicket and BusTicket as children, inheriting the Ticket. `@Inheritance` allows to set the way how to store parents and children in db. With `InheritanceType.SINGLE_TABLE` all the inheritors of the Ticket will be stored in the same table. A discriminator column called `type`, will be used to distinguish between the different subclasses.

Reply:
Cool, also would be great to read/learn about other inheritance types and get perspective in which cases need to choose this or that type